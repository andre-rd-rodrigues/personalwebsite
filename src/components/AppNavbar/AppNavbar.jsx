import AppIcon from "components/AppIcon/AppIcon";
import React from "react";
import { useState } from "react";
import { Nav, Navbar, Offcanvas } from "react-bootstrap";
import { Link } from "react-router-dom";
import styles from "./appnavbar.module.scss";
import FeatherIcon from "feather-icons-react";

const AppNavbar = () => {
  const [show, setShow] = useState(false);

  return (
    <Navbar expand="lg" fixed="top" className={styles.nav}>
      <Navbar.Brand as={Link} to="/">
        <div className={styles.logo}>
          <div
            id="personal_picture"
            style={{
              background: `url(${
                process.env.PUBLIC_URL + "images/profile.png"
              }) center center`
            }}
          />
          <h1>AR</h1>
          <p>Programação</p>
        </div>
      </Navbar.Brand>
      <Navbar.Toggle
        aria-controls="offcanvas-container"
        onClick={() => setShow(true)}
      >
        <AppIcon />
        <FeatherIcon icon="menu" color="white" />
      </Navbar.Toggle>
      <Navbar.Offcanvas
        show={show}
        responsive="lg"
        onHide={() => setShow(false)}
        aria-labelledby="offcanvas-container"
        placement="end"
      >
        <Offcanvas.Header closeButton />
        <Offcanvas.Body>
          <Nav className={styles.offcanvasLinks}>
            <Nav.Link href="/works">Trabalhos</Nav.Link>
            <br />
            <Nav.Link href="/blog">Blog</Nav.Link> <br />
            <Nav.Link href="/about">Sobre</Nav.Link> <br />
            <Nav.Link href="/contact">Contactos</Nav.Link> <br />
          </Nav>
        </Offcanvas.Body>
      </Navbar.Offcanvas>
    </Navbar>
  );
};

export default AppNavbar;
