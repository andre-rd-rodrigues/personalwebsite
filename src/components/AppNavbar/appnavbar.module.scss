@import "assets/styles/mixins";

.nav {
  top: 0px;
  padding: 15px var(--padding-y) !important;
  background-color: $dark;
  h1,
  p,
  a {
    color: $white !important;
  }

  button[aria-controls="offcanvas-container"] {
    border: none;
  }
}

.logo {
  display: flex;
  align-items: center;

  div[id="personal_picture"] {
    background: url("https://scontent.flis8-1.fna.fbcdn.net/v/t1.6435-9/83004576_2502375390091091_6509502809489014784_n.jpg?_nc_cat=104&ccb=1-7&_nc_sid=09cbfe&_nc_ohc=DLwHg_owZ3kAX_SGG9A&_nc_ht=scontent.flis8-1.fna&oh=00_AT9T70XVlZ8IA-_T-zl2wxvdkLDVLXzpaOYNndJFnhKqUw&oe=6302B5FB")
      center center no-repeat;
    background-size: cover;
    width: 50px;
    height: 50px;
    border-radius: 18px;
  }

  h1,
  p {
    margin: 0;
  }
  h1 {
    font-family: Rozha;
    font-weight: bold;
    font-size: 32px;
    margin-left: 15px;
  }
  p {
    font-size: 1rem;
    display: flex;
    align-items: center;
    &::before {
      content: "";
      display: flex;
      height: 1px;
      width: 50px;
      background: white;
      margin: 0 10px;
    }
  }
}

//Offcanvas
.offcanvasLinks {
  transform: translateY(30px);

  a {
    @include hover_underline($white);
  }
}
div[class="offcanvas offcanvas-end"] {
  height: 100px;
}

div[class="offcanvas offcanvas-end show"],
div[class="offcanvas offcanvas-end offcanvas-toggling"],
div[class="offcanvas offcanvas-end offcanvas-toggling show"] {
  background-color: $light-grey !important;
  max-width: 320px;
  a {
    color: $dark;
    opacity: 0.8;
    font-weight: 400;
    font-size: var(--fs-l-s);
    text-align: end;
    font-family: Italiana !important;
    transition: 0.4s ease;
    background-image: none;
    &:hover {
      color: $yellow;
      opacity: 1;
      letter-spacing: 4px;
    }
  }

  button[class="btn-close"] {
    transform: scale(0.7);
  }
}
div[class="offcanvas-header"] {
  justify-content: end;
}
