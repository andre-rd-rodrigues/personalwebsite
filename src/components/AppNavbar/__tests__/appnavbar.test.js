import { render } from "utils/tests/test-utils";
import AppNavbar from "../AppNavbar";

describe("Portfolio Category Page", () => {
  it("renders page correctly", () => {
    const { asFragment } = render(<AppNavbar />);

    expect(asFragment()).toMatchSnapshot();
  });
});
