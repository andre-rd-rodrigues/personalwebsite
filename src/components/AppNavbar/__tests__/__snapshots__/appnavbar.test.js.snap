// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Category Page renders page correctly 1`] = `
<DocumentFragment>
  <nav
    class="nav navbar navbar-expand-lg navbar-light fixed-top"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <div
        class="logo"
      >
        <div
          id="personal_picture"
        />
        <h1>
          AR
        </h1>
        <p>
           Web Developer
        </p>
      </div>
    </a>
    <button
      aria-controls="offcanvas-container"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <div
        class="app-icon"
        role="button"
        style="cursor: pointer; display: inline;"
        tabindex="0"
      >
        <svg
          class="feather feather-menu "
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="stroke-width: 1;"
          viewBox="0 0 24 24"
          width="20"
        >
          <g>
            <line
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </g>
        </svg>
      </div>
    </button>
  </nav>
</DocumentFragment>
`;
